java-viewer/src/main/java/org/sonar/java/cfg/CFGDebug.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * SonarQube Java
+ * Copyright (C) 2012-2017 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.java.cfg;
+
+import org.sonar.java.ast.SyntaxTreeDebug;
+import org.sonar.java.cfg.CFG.Block;
+import org.sonar.plugins.java.api.tree.Tree;
+import org.sonar.plugins.java.api.tree.Tree.Kind;
+
+public class CFGDebug {
+
+  private static final int MAX_KINDNAME = Kind.UNSIGNED_RIGHT_SHIFT_ASSIGNMENT.name().length() + 5;
+
+  private CFGDebug() {
+  }
+
+  public static String toString(CFG cfg) {
+    StringBuilder buffer = new StringBuilder();
+    buffer.append("Starts at B");
+    buffer.append(cfg.entry().id());
+    buffer.append('\n');
+    buffer.append('\n');
+    for (Block block : cfg.blocks()) {
+      buffer.append(toString(block));
+    }
+    return buffer.toString();
+  }
+
+  private static String toString(CFG.Block block) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 17 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3776)